### PR TITLE
Properly check an outgoing call status

### PR DESCRIPTION
**Pull request by sdex**

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/dialer/extensions/Call.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import android.os.Build
+import android.telecom.Call
+import android.telecom.Call.STATE_CONNECTING
+import android.telecom.Call.STATE_DIALING
+import android.telecom.Call.STATE_SELECT_PHONE_ACCOUNT
+
+private val OUTGOING_CALL_STATES = arrayOf(STATE_CONNECTING, STATE_DIALING, STATE_SELECT_PHONE_ACCOUNT)
+
+fun Call.getStateCompat(): Int {
+    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/dialer/extensions/Call.kt:12]: use isSPlus() here

File: app/src/main/kotlin/com/simplemobiletools/dialer/extensions/Call.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        details.state
+    } else {
+        @Suppress("DEPRECATION")
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/dialer/extensions/Call.kt:15]: pur such suppressions above the function itself, not to the code

File: app/src/main/kotlin/com/simplemobiletools/dialer/extensions/Call.kt (hunk 1 above)
Comment (sdex) [reply app/src/main/kotlin/com/simplemobiletools/dialer/extensions/Call.kt]: done

**Comment by tibbi** [comment]
looks good, thanks

**Pull request merged**